rosbag2_transport/src/rosbag2_transport/topic_filter.cpp

```diff
@@ -121,11 +121,6 @@ bool TopicFilter::take_topic(
     return false;
   }
 
-  const std::string & topic_type = topic_types[0];
```

Comment:
I know this is the problem that users are complaining about right now, but should we also move the `has_single_type` call to after selection, for similar reasons?

Reply:
@clalancette Good catch and make sense to me. I will do.

Reply:
@clalancette Done.